src/mono/wasm/features.md

```diff
@@ -2,36 +2,125 @@
+
+## fetch browser API
 
-# fetch browser API
 Is required if the application uses [HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient)
+
 NodeJS needs to install `node-fetch` and `node-abort-controller` npm packages.
 
-# WebSocket browser API
+## WebSocket browser API
+
 Is required if the application uses [WebSocketClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.websockets.clientwebsocket)
+
 NodeJS needs to install `ws` npm package.
 
+## Content security policy
+
+dotnet runtime for wasm is CSP compliant starting from .Net 8, except legacy JS interop methods.
+
+In order to enable it, please set HTTP headers similar to `Content-Security-Policy: default-src 'self' 'wasm-unsafe-eval'`
+
+See also [CSP on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP)
+
+## ICU
+
+_TODO_
+
+## Timezones
+
+Browsers don't offer API for working with time zone database and so we have to bring the time zone data as part of the application.
+
+If your application doesn't need to work with TZ, you can reduce download size by `<InvariantTimezone>true</InvariantTimezone>`.
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
+
 # Shell environments - NodeJS & V8
 We pass most of the unit tests with NodeJS v 14 but it's not fully supported target platform. We would like to hear about community use-cases.
-We also use v8 engine version 11 or higher to run some of the tests. The engine is lacking many APIs and features.
\ No newline at end of file
+
+We also use v8 engine version 11 or higher to run some of the tests. The engine is lacking many APIs and features.
+
+# Mobile phones
+
+Recent mobile phones have browser integration which could be upgraded separately from the operating system.
+
+Also note that all browsers on iOS are just wrapper for Safari engine and so it's limited by Safari features and version, rather than wrapper version and brand.
+
+Mobile phones usually have limited resources. Memory and download speed are major concern.
+
+## Resources consumed on the target device
+
+dotnet is complex and large application, it consists of
+- dotnet runtime, including garbage collector, IL interpreter and browser specific JIT
+- dotnet base class library
+- emulation layer which is bringing missing features of the OS, which the browser doesn't provide. Like timezone database or ICU.
+- integration with the browser JavaScript APIs, for example HTTP and WebSocket client
+- application code
+
+All of the mentioned code and data need to be downloaded and loaded into the browser memory during the dotnet startup sequence.
+Browser itself will run JIT compilation of the WASM and JS code, which consumes memory and CPU cycles too.
+
+## WASM linear memory
+
+You can override initial size of the WASM linear memory by `<EmccInitialHeapSize>16777216</EmccInitialHeapSize>`.
+This requires that you have [wasm-tools workload](#wasm-tools-workload) installed.
```

Comment:
We could also explain briefly here why you might choose to do this, i.e. `Setting an initial size based on how much memory your application typically uses will reduce the number of times the heap needs to grow, which may enable it to run on devices with lower memory`. I assume that's the reason for doing this, anyway.